src/universal/mutations/RemoveReflectionMutation.js

```diff
@@ -33,32 +34,22 @@ const mutation = graphql`
   }
 `;
 
-export const removeReflectionUpdater = (payload: RecordProxy, store: RecordSourceSelectorProxy) => {
-  // TODO - maeby https://github.com/dan-f/maeby
-  if (!payload) {
-    return;
-  }
-  const payloadMeeting = payload.getLinkedRecord('meeting');
-  if (!payloadMeeting) {
-    return;
-  }
-  const meetingId = payloadMeeting.getValue('id');
-  const reflection = payload.getLinkedRecord('reflection');
-  if (!reflection) {
-    return;
-  }
-  const meeting = store.get(meetingId);
-  if (!meeting) {
-    return;
-  }
-  const reflections = meeting.getLinkedRecords('reflections');
-  if (!reflections) {
-    return;
-  }
-  const newReflections = reflections.filter((r) => (
-    r.getValue('id') !== reflection.getValue('id')
-  ));
-  meeting.setLinkedRecords(newReflections, 'reflections');
+export const removeReflectionUpdater = (payload: ?RecordProxy, store: RecordSourceSelectorProxy) => {
```

Comment:
Yeah I personally find this readable because it's not bogged down with the `if (!thing) { return }` statements. I also dig that `getInProxy` pattern, which is cool, and I think the two patterns can be combined! As in, in the example you gave, `agendaItemId` can still be null, so you should actually be checking `if (!agendaItemId) { return }` before calling the handler. So perhaps I can change this to use the `getInProxy` helpers to traverse the path to the final document, and then wrap those optional types in `maybe`.

Reply:
since that handler could be used anywhere, it needs to check for null anyways, so i just do it there.

Reply:
Yeah, I'd like to keep this as is. That `getInProxy` thing is hardcoded to do a scalar lookup at the end of its path, which isn't general enough for use here. Also, the code works as-is, and is shorter than manually checking `undefined` at each step.

Reply:
The motivation here isn't using `maybe` for `maybe`'s sake, it's to make the code safer and more correct to avoid production crashes. I've seen enough `TypeError: <x> is undefined`'s at this point to warrant a more general approach to dealing with optional types.